src/huggingface_hub/repocard.py

```diff
@@ -24,15 +25,26 @@ def metadata_load(local_path: Union[str, Path]) -> Optional[Dict]:
 
 
 def metadata_save(local_path: Union[str, Path], data: Dict) -> None:
-    data_yaml = yaml.dump(data, sort_keys=False)
-    # sort_keys: keep dict order
-    content = Path(local_path).read_text() if Path(local_path).is_file() else ""
-    match = REGEX_YAML_BLOCK.search(content)
-    if match:
-        output = (
-            content[: match.start()] + f"---\n{data_yaml}---\n" + content[match.end() :]
-        )
-    else:
-        output = f"---\n{data_yaml}---\n{content}"
-
-    Path(local_path).write_text(output)
+    # try to preserve newlines
+    linebrk = "\n"
+    # this is known not to work with ^M linebreaks, so ^M are replaced by \n
+    with open(local_path, "r") as readme:
```

Comment:
this first open is meant to ensure we detect the real newline character in the file.
and to read its contents

Reply:
```suggestion
    with open(local_path, "r", newline="") as readme:
```